### ISSUE TITLE
A sortKeys Option Would be Very Handy

### ISSUE THREAD

**Issue by machineghost** [logi/chai-diff#2]
First off let me say thanks so much for making this plugin: having specific diff output makes such a huge difference when (for instance) two objects with 20+ keys are different by only a single key.

However, one minor improvement could make `not.to.be.differentFrom` the new (and improved) `to.eql`: key sorting.  Currently if I compare

    expect({ a: 1, b: 2 }).not.to.be.differentFrom({ b: 2, a: 1 });

I will get an expectation failure.  It'd be really great if I could avoid one by doing:

    expect({ a: 1, b: 2 }).not.to.be.differentFrom({ b: 2, a: 1 }, { sortKeys: true });

In fact, I'd even argue that `sortKeys` should default to true: 99% of the time when you compare objects in Javascript you don't care what order their keys are in.

Thanks for considering it.

P.S. A basic implementation using Lodash/Underscore's `isObject` (which could easily be removed) would be:

    const sortKeys = (obj) => {
      if (!_.isObject(obj)) return obj;
    
      const newObj = {};
      Object.keys(obj).sort().forEach(key => Object.assign(newObj, { [key]: sortKeys(obj[key]) }));
      return newObj;
    };

    expect(sortKeys(actualData)).to.not.to.be.differentFrom(sortKeys(expectedData));

**Comment by logi** [comment]
I fully agree. However, I've left the position where I had the itch that this was made to scratch, and am not doing javascript development at all, so the context switch would be too much for me to bother with. But if someone(...) makes a pull request which sorts keys by default and, if you like, allows that to be turned off, then I'll merge it in.

(I would have responded earlier, but it's been Easter/Thingyan/Songkran)
